test/conversions.jl

```diff
@@ -114,15 +163,93 @@ end
+    @test promote_type(ARGB32, GrayA) === ARGB
+
+    @test promote_type(AbstractRGB, Gray{Float16}) === RGB
+    @test promote_type(RGB, AbstractGray{Float16}) === RGB
+    @test promote_type(AbstractRGB{Float16}, Gray{Float16}) === RGB{Float16}
+    @test promote_type(AbstractRGB{Float16}, Gray24) === RGB{Float32}
+
+    @test promote_type(AbstractGray, RGB{Float16}) === RGB
+    @test promote_type(Gray, AbstractRGB{Float16}) === RGB
+    @test promote_type(AbstractGray{Float16}, RGB{Float16}) === RGB{Float16}
+    @test promote_type(AbstractGray{Float16}, RGB24) === RGB{Float32}
+
+    @test promote_type(AbstractRGB, AGray{Float16}) === ARGB
+    @test promote_type(RGB, AbstractAGray{Gray{Float16},Float16}) === ARGB
+    @test promote_type(AbstractRGB{Float16}, AGray{Float16}) === ARGB{Float16}
+    @test promote_type(AbstractRGB{Float16}, AGray32) === ARGB{Float32}
+
+    @test promote_type(AbstractGray, ARGB{Float16}) === ARGB
+    @test promote_type(Gray, AbstractARGB{RGB{Float16},Float16}) === ARGB
+    @test promote_type(AbstractGray{Float16}, ARGB{Float16}) === ARGB{Float16}
+    @test promote_type(AbstractGray{Float16}, ARGB32) === ARGB{Float32}
+
+    # @test promote_type(AbstractARGB, AGray{Float16}) === ARGB
+    # @test promote_type(ARGB, AbstractAGray{Gray{Float16},Float16}) === ARGB
+    # @test promote_type(AbstractARGB{RGB{Float16},Float16}, AGray{Float16}) === ARGB{Float16}
+    # @test promote_type(AbstractARGB{RGB{Float16},Float16}, AGray32) === ARGB{Float32}
+end
+
+@testset "promotions between different spaces" begin
+    @test promote_type(RGB, HSV) == Color3
+    @test promote_type(RGB, HSV{Float16}) == Color3
+    @test promote_type(HSV, RGB{Float16}) == Color3
+
+    @test promote_type(RGB, HSVA) == ColorAlpha{C,T,4} where {T, C<:Color3}
+    @test promote_type(AHSV, RGB) == AlphaColor{C,T,4} where {T, C<:Color3}
+    @test promote_type(RGB, HSVA{Float16}) == ColorAlpha{C,T,4} where {T, C<:Color3}
+    @test promote_type(AHSV, RGB{Float16}) == AlphaColor{C,T,4} where {T, C<:Color3}
+
+    @test promote_type(HSV, RGBA) == ColorAlpha{C,T,4} where {T, C<:Color3}
+    @test promote_type(ARGB, HSV) == AlphaColor{C,T,4} where {T, C<:Color3}
+    @test promote_type(HSV, RGBA{Float16}) == ColorAlpha{C,T,4} where {T, C<:Color3}
+    @test promote_type(ARGB, HSV{Float16}) == AlphaColor{C,T,4} where {T, C<:Color3}
+
+    @test promote_type(HSVA, RGBA) == ColorAlpha{C,T,4} where {T, C<:Color3}
+    @test promote_type(ARGB, AHSV) == AlphaColor{C,T,4} where {T, C<:Color3}
+    @test promote_type(AHSV, RGBA) == Transparent3
+    @test promote_type(ARGB, HSVA) == Transparent3
+
+    @test promote_type(RGB24, HSV) == Color3
+    @test promote_type(RGB24, HSVA{Float16}) == ColorAlpha{C,Float32,4} where C<:Color3{Float32}
+    @test promote_type(ARGB32, HSV) == AlphaColor{C,T,4} where {T, C<:Color3}
+    @test promote_type(ARGB32, HSVA{Float16}) == TransparentColor{C,Float32,4} where C<:Color3{Float32}
```

Comment:
Do we need any tests where both are parametric? AFAICT it should be handled fine (these are actually the harder cases), but just wondering. (Optional.)

Reply:
I just forgot because I was so full of non-concrete types.:smile:
I'll add tests.